changes/2152.fix.md

```diff
@@ -0,0 +1 @@
+Do early null-check on data fetched from DB in `check_presets` API function and avoid an infinite recursion in property method in client.APIConfig
```

Comment:
Please rewrite the news fragment in a way of expressing the changes this PR has introduced on a more consumer-specific way (e.g. what kind of quirky behavior this PR has fixed).

Reply:
I updated it for some clients who use manager APIs and SDK